### PR TITLE
Fix activity heat map padding & locale

### LINKED ISSUE #30808: broken styles on hover in activity
### Description

image

both on main screen and profile

### Gitea Version

trunk

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

docker

### Database

None

### PR DESCRIPTION
**Pull request by wxiaoguang**
Fix #30808

image

image

### CODE REVIEW

**Comment by wxiaoguang** [comment]
> Can you copy the CSS from #30822 into here? I think the tooltip style is better for this tippy.

Done, and the screenshots are updated.

**Comment by wxiaoguang** [comment]
Also added the border radius, looks better.

**Comment by wxiaoguang** [comment]
> Your border override is not working, it should have no visible border.

OK, I see the problem. `border none` should be on `tippy-box` while others should be on `tippy-content` (otherwise there will be stranage animation)

**Comment by silverwind** [comment]
Imho, put all styles on `tippy-box`.

**Comment by wxiaoguang** [comment]
> Imho, put all styles on `tippy-box`. Other themes do it like that too.

Nope, you could take a try, it would case strange animations.

File: web_src/js/components/ActivityHeatmap.vue [hunk 1]
```diff
@@ -55,15 +55,33 @@ export default {
...
     :values="values"
     :range-color="colorRange"
     @day-click="handleDayClick($event)"
   />
 </template>
+<style>
+/* A quick patch for vue3-calendar-heatmap's tooltip padding, to avoid conflicting with other tippy contents.
+At the moment we could only identify the tooltip by its transition property.
+https://github.com/razorness/vue3-calendar-heatmap/blob/955626176cb5dc3d3ead8120475c2e5e753cc392/src/components/CalendarHeatmap.vue#L202
+This selector should be replaced by a more specific one if the library adds a CSS class.
```
Comment (silverwind) [inline web_src/js/components/ActivityHeatmap.vue]:
```suggestion
This selector should be replaced by a more specific one if the library adds a CSS class.
TODO: Remove this after https://github.com/razorness/vue3-calendar-heatmap/pull/34 is merged and released.
```

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
> > Imho, put all styles on `tippy-box`. Other themes do it like that too.
> 
> Nope, you could take a try, it would case strange animations.
> 
> image
> 
> image

Hmm it looked alright in my PR at least.

**Comment by wxiaoguang** [comment]
> Hmm it looked alright in my PR at least.

I think the problem is caused by the `transition: transform 0.1s ease-out` selector. 

Pros: it won't conflict with other tippy contents.
Cons: somewhat hacky and need to use 2 selectors.

(I added one more comment "after PR 34 ..  then all styles could be merged into `.tippy-box` selector.")

**Comment by silverwind** [comment]
`.tippy-box:not([data-theme])` will work, no other tippy will match on it as they will either have a custom theme or [`default`](https://github.com/go-gitea/gitea/blob/6ff2acc52c976e9d7bb6a5693f8a2365d12400f5/web_src/js/modules/tippy.js#L43).

**Comment by wxiaoguang** [comment]
> `.tippy-box:not([data-theme])` will work, no other tippy will match on it as they will either have a custom theme or [`default`](https://github.com/go-gitea/gitea/blob/6ff2acc52c976e9d7bb6a5693f8a2365d12400f5/web_src/js/modules/tippy.js#L43).

I do not think it is good/right to make it potential conflict with other tippy contents (we can never assume that every tippy content is managed by us, there would be more 3rd libraries)

But since you insist, 5f229c813f81a68952c0cf2ef21e9171111dd653

**Comment by wxiaoguang** [comment]
> I just published my fork to npm, we can use it until upstream merges my PR:
> 
> `@silverwind/vue3-calendar-heatmap@2.0.6`

That sounds good, could you help to  update this PR?

**Comment by silverwind** [comment]
I will try use my fork here.

**Comment by wxiaoguang** [comment]
Awesome, perfect!

**Review by silverwind — APPROVED** [review]

**Pull request merged**